pkg/cli/validate_test.go

```diff
@@ -0,0 +1,265 @@
+	cases := []struct {
+		name   string
+		config *config.CLIConfig
+		args   []string
+		pipe   bool
+		expOut string
+		expErr string
+	}{
+		{
+			name:   "too_many_args",
+			args:   []string{"foo"},
+			expErr: `accepts 0 arg(s)`,
+		},
+		{
+			name:   "missing_token",
+			args:   nil,
+			expErr: `"token" not set`,
+		},
+		{
+			name: "invalid_token",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args:   []string{"-t=invalid token"},
+			expErr: `failed to parse token headers`,
+		},
+		{
+			name: "signed",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", signedToken},
+			expOut: `-------RESULT-------
+validated!
+breakglass  false
+----JUSTIFICATION----
+explanation  "test"
+foo          "bar"
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+		{
+			name: "breakglass",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", breakglassToken},
+			expOut: `-------RESULT-------
+validated!
+breakglass  true
+----JUSTIFICATION----
+breakglass  "prod is down"
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+		{
+			name: "signed_from_pipe",
+			config: &config.CLIConfig{
+				JWKSEndpoint: svr.URL + path,
+			},
+			args: []string{"-t", "-"},
+			pipe: true,
+			expOut: `-------RESULT-------
+validated!
+breakglass  false
+----JUSTIFICATION----
+explanation  "test"
+foo          "bar"
+---STANDARD CLAIMS---
+aud  ["dev.abcxyz.jvs"]
+iat  "1970-01-01T00:00:00Z"
+iss  "jvsctl"
+jti  "test-jwt"
+nbf  "1970-01-01T00:00:00Z"
+sub  "jvsctl"
+`,
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			cmd := newValidateCmd(tc.config)
+			var stdout string
+			var gotErr error
+			if tc.pipe {
+				// create tempFile with token
+				tempFile, err := os.CreateTemp("", "validate_test*")
+				if err != nil {
+					t.Errorf("failed to create temp file: %s", err)
+				}
+				// remove tempFile
+				defer os.Remove(tempFile.Name())
+
+				if _, err := tempFile.Write([]byte(signedToken)); err != nil {
+					t.Errorf("failed to write to temp file: %s", err)
+				}
+				if _, err := tempFile.Seek(0, 0); err != nil {
+					t.Errorf("failed to seek to the beginning of the temp file: %s", err)
+				}
+
+				stdout, _, gotErr = testExecuteCommandStdin(t, cmd, tempFile, tc.args...)
+				if err := tempFile.Close(); err != nil {
+					t.Errorf("failed to close temp file: %s", err)
+				}
+			} else {
+				stdout, _, gotErr = testExecuteCommand(t, cmd, tc.args...)
+			}
+			if diff := testutil.DiffErrString(gotErr, tc.expErr); diff != "" {
+				t.Fatal(diff)
+			}
+			if gotErr != nil {
+				return
+			}
+
+			if diff := cmp.Diff(tc.expOut, stdout); diff != "" {
```

Comment:
Do we capture the error output? Should we compare them?

Reply:
Is it necessary? the error output is the token flag usage.

Reply:
You're right. `DiffErrString` already does that.